application/Validation.go

```diff
@@ -0,0 +1,46 @@
+package application
+
+import (
+	"authz/domain"
+	"regexp"
+
+	"github.com/go-playground/validator/v10"
+	"github.com/golang/glog"
+)
+
+var validatorInstance *validator.Validate
+
+// ValidateEvent performs validation on the provided event struct and returns true if the struct is valid, else it returns false and an error message object
+func ValidateEvent(evt interface{}) (bool, error) {
+	if validatorInstance == nil {
+		initializeValidator() //TODO: move to startup?
```

Comment:
Moving it to startup is perhaps giving the validator too much respect. :) It's just a validator, and I think the fact it's a singleton achieves the objective (assuming creating new validators really has a cost).

Reply:
same here. I am not sure it's worth the additional code. Depends a bit on how often this will change. It'd be good to have an injectable validator to improve the testability when we introduce "new" custom validations or change existing ones I think, so maybe worth it. Not hard on it at all, though